[2023-10-11T19:27:42.401+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-02T00:00:00+00:00 [queued]>
[2023-10-11T19:27:42.416+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-02T00:00:00+00:00 [queued]>
[2023-10-11T19:27:42.418+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 3
[2023-10-11T19:27:42.438+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): trusted> on 2023-10-02 00:00:00+00:00
[2023-10-11T19:27:42.445+0000] {standard_task_runner.py:57} INFO - Started process 560 to run task
[2023-10-11T19:27:42.448+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'wikipedia_etl', 'trusted', 'scheduled__2023-10-02T00:00:00+00:00', '--job-id', '19', '--raw', '--subdir', 'DAGS_FOLDER/wikipedia_etl.py', '--cfg-path', '/tmp/tmpvrmnmzj6']
[2023-10-11T19:27:42.452+0000] {standard_task_runner.py:85} INFO - Job 19: Subtask trusted
[2023-10-11T19:27:42.519+0000] {task_command.py:415} INFO - Running <TaskInstance: wikipedia_etl.trusted scheduled__2023-10-02T00:00:00+00:00 [running]> on host f711e14c143e
[2023-10-11T19:27:42.703+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='wikipedia_etl' AIRFLOW_CTX_TASK_ID='trusted' AIRFLOW_CTX_EXECUTION_DATE='2023-10-02T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-10-02T00:00:00+00:00'
[2023-10-11T19:28:37.946+0000] {local_task_job_runner.py:294} WARNING - State of this instance has been externally set to None. Terminating instance.
[2023-10-11T19:28:37.948+0000] {process_utils.py:131} INFO - Sending 15 to group 560. PIDs of all processes in the group: [561, 560]
[2023-10-11T19:28:37.948+0000] {process_utils.py:86} INFO - Sending the signal 15 to group 560
[2023-10-11T19:28:37.949+0000] {taskinstance.py:1630} ERROR - Received SIGTERM. Terminating subprocesses.
[2023-10-11T19:28:37.958+0000] {taskinstance.py:1935} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/wikipedia_etl.py", line 133, in trusted
    sc = SparkContext(conf=conf)
  File "/home/airflow/.local/lib/python3.8/site-packages/pyspark/context.py", line 201, in __init__
    SparkContext._ensure_initialized(self, gateway=gateway, conf=conf)
  File "/home/airflow/.local/lib/python3.8/site-packages/pyspark/context.py", line 436, in _ensure_initialized
    SparkContext._gateway = gateway or launch_gateway(conf)
  File "/home/airflow/.local/lib/python3.8/site-packages/pyspark/java_gateway.py", line 104, in launch_gateway
    time.sleep(0.1)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1632, in signal_handler
    raise AirflowException("Task received SIGTERM signal")
airflow.exceptions.AirflowException: Task received SIGTERM signal
[2023-10-11T19:28:37.961+0000] {taskinstance.py:1398} INFO - Marking task as UP_FOR_RETRY. dag_id=wikipedia_etl, task_id=trusted, execution_date=20231002T000000, start_date=20231011T192742, end_date=20231011T192837
[2023-10-11T19:28:37.973+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 19 for task trusted ((psycopg2.errors.ForeignKeyViolation) insert or update on table "task_fail" violates foreign key constraint "task_fail_ti_fkey"
DETAIL:  Key (dag_id, task_id, run_id, map_index)=(wikipedia_etl, trusted, scheduled__2023-10-02T00:00:00+00:00, -1) is not present in table "task_instance".

[SQL: INSERT INTO task_fail (task_id, dag_id, run_id, map_index, start_date, end_date, duration) VALUES (%(task_id)s, %(dag_id)s, %(run_id)s, %(map_index)s, %(start_date)s, %(end_date)s, %(duration)s) RETURNING task_fail.id]
[parameters: {'task_id': 'trusted', 'dag_id': 'wikipedia_etl', 'run_id': 'scheduled__2023-10-02T00:00:00+00:00', 'map_index': -1, 'start_date': datetime.datetime(2023, 10, 11, 19, 27, 42, 402330, tzinfo=Timezone('UTC')), 'end_date': datetime.datetime(2023, 10, 11, 19, 28, 37, 961027, tzinfo=Timezone('UTC')), 'duration': 55}]
(Background on this error at: https://sqlalche.me/e/14/gkpj); 560)
[2023-10-11T19:28:38.001+0000] {process_utils.py:79} INFO - Process psutil.Process(pid=560, status='terminated', exitcode=1, started='19:27:41') (560) terminated with exit code 1
[2023-10-11T19:28:38.002+0000] {process_utils.py:79} INFO - Process psutil.Process(pid=561, status='terminated', started='19:27:42') (561) terminated with exit code None
[2023-10-11T19:32:00.760+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-02T00:00:00+00:00 [queued]>
[2023-10-11T19:32:00.767+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-02T00:00:00+00:00 [queued]>
[2023-10-11T19:32:00.767+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 3
[2023-10-11T19:32:00.778+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): trusted> on 2023-10-02 00:00:00+00:00
[2023-10-11T19:32:00.782+0000] {standard_task_runner.py:57} INFO - Started process 1227 to run task
[2023-10-11T19:32:00.784+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'wikipedia_etl', 'trusted', 'scheduled__2023-10-02T00:00:00+00:00', '--job-id', '26', '--raw', '--subdir', 'DAGS_FOLDER/wikipedia_etl.py', '--cfg-path', '/tmp/tmp4wuu81k0']
[2023-10-11T19:32:00.787+0000] {standard_task_runner.py:85} INFO - Job 26: Subtask trusted
[2023-10-11T19:32:00.822+0000] {task_command.py:415} INFO - Running <TaskInstance: wikipedia_etl.trusted scheduled__2023-10-02T00:00:00+00:00 [running]> on host f711e14c143e
[2023-10-11T19:32:00.879+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='wikipedia_etl' AIRFLOW_CTX_TASK_ID='trusted' AIRFLOW_CTX_EXECUTION_DATE='2023-10-02T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-10-02T00:00:00+00:00'
[2023-10-11T19:32:16.203+0000] {python.py:194} INFO - Done. Returned value was: None
[2023-10-11T19:32:16.216+0000] {taskinstance.py:1398} INFO - Marking task as SUCCESS. dag_id=wikipedia_etl, task_id=trusted, execution_date=20231002T000000, start_date=20231011T193200, end_date=20231011T193216
[2023-10-11T19:32:16.237+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2023-10-11T19:32:16.249+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2023-10-11T19:46:46.046+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-02T00:00:00+00:00 [queued]>
[2023-10-11T19:46:46.053+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-02T00:00:00+00:00 [queued]>
[2023-10-11T19:46:46.053+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 3
[2023-10-11T19:46:46.063+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): trusted> on 2023-10-02 00:00:00+00:00
[2023-10-11T19:46:46.069+0000] {standard_task_runner.py:57} INFO - Started process 3835 to run task
[2023-10-11T19:46:46.072+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'wikipedia_etl', 'trusted', 'scheduled__2023-10-02T00:00:00+00:00', '--job-id', '46', '--raw', '--subdir', 'DAGS_FOLDER/wikipedia_etl.py', '--cfg-path', '/tmp/tmp0a76l9qi']
[2023-10-11T19:46:46.074+0000] {standard_task_runner.py:85} INFO - Job 46: Subtask trusted
[2023-10-11T19:46:46.111+0000] {task_command.py:415} INFO - Running <TaskInstance: wikipedia_etl.trusted scheduled__2023-10-02T00:00:00+00:00 [running]> on host f711e14c143e
[2023-10-11T19:46:46.180+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='wikipedia_etl' AIRFLOW_CTX_TASK_ID='trusted' AIRFLOW_CTX_EXECUTION_DATE='2023-10-02T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-10-02T00:00:00+00:00'
[2023-10-11T19:47:17.230+0000] {python.py:194} INFO - Done. Returned value was: None
[2023-10-11T19:47:17.239+0000] {taskinstance.py:1398} INFO - Marking task as SUCCESS. dag_id=wikipedia_etl, task_id=trusted, execution_date=20231002T000000, start_date=20231011T194646, end_date=20231011T194717
[2023-10-11T19:47:17.277+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2023-10-11T19:47:17.290+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2023-10-13T17:17:17.124+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-02T00:00:00+00:00 [queued]>
[2023-10-13T17:17:17.137+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-02T00:00:00+00:00 [queued]>
[2023-10-13T17:17:17.137+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 3
[2023-10-13T17:17:17.152+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): trusted> on 2023-10-02 00:00:00+00:00
[2023-10-13T17:17:17.159+0000] {standard_task_runner.py:57} INFO - Started process 362 to run task
[2023-10-13T17:17:17.163+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'wikipedia_etl', 'trusted', 'scheduled__2023-10-02T00:00:00+00:00', '--job-id', '6', '--raw', '--subdir', 'DAGS_FOLDER/wikipedia_etl.py', '--cfg-path', '/tmp/tmppq52y45v']
[2023-10-13T17:17:17.166+0000] {standard_task_runner.py:85} INFO - Job 6: Subtask trusted
[2023-10-13T17:17:17.220+0000] {task_command.py:415} INFO - Running <TaskInstance: wikipedia_etl.trusted scheduled__2023-10-02T00:00:00+00:00 [running]> on host feba906f3341
[2023-10-13T17:17:17.345+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='wikipedia_etl' AIRFLOW_CTX_TASK_ID='trusted' AIRFLOW_CTX_EXECUTION_DATE='2023-10-02T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-10-02T00:00:00+00:00'
[2023-10-13T17:18:00.016+0000] {python.py:194} INFO - Done. Returned value was: None
[2023-10-13T17:18:00.036+0000] {taskinstance.py:1398} INFO - Marking task as SUCCESS. dag_id=wikipedia_etl, task_id=trusted, execution_date=20231002T000000, start_date=20231013T171717, end_date=20231013T171800
[2023-10-13T17:18:00.102+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2023-10-13T17:18:00.128+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2023-10-16T11:25:08.677+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-02T00:00:00+00:00 [queued]>
[2023-10-16T11:25:08.684+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-02T00:00:00+00:00 [queued]>
[2023-10-16T11:25:08.684+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 3
[2023-10-16T11:25:08.696+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): trusted> on 2023-10-02 00:00:00+00:00
[2023-10-16T11:25:08.703+0000] {standard_task_runner.py:57} INFO - Started process 336 to run task
[2023-10-16T11:25:08.707+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'wikipedia_etl', 'trusted', 'scheduled__2023-10-02T00:00:00+00:00', '--job-id', '6', '--raw', '--subdir', 'DAGS_FOLDER/wikipedia_etl.py', '--cfg-path', '/tmp/tmpghry_sr3']
[2023-10-16T11:25:08.710+0000] {standard_task_runner.py:85} INFO - Job 6: Subtask trusted
[2023-10-16T11:25:08.750+0000] {task_command.py:415} INFO - Running <TaskInstance: wikipedia_etl.trusted scheduled__2023-10-02T00:00:00+00:00 [running]> on host ab44499fca46
[2023-10-16T11:25:08.817+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='wikipedia_etl' AIRFLOW_CTX_TASK_ID='trusted' AIRFLOW_CTX_EXECUTION_DATE='2023-10-02T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-10-02T00:00:00+00:00'
[2023-10-16T11:25:44.891+0000] {python.py:194} INFO - Done. Returned value was: None
[2023-10-16T11:25:44.906+0000] {taskinstance.py:1398} INFO - Marking task as SUCCESS. dag_id=wikipedia_etl, task_id=trusted, execution_date=20231002T000000, start_date=20231016T112508, end_date=20231016T112544
[2023-10-16T11:25:44.958+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2023-10-16T11:25:44.988+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2023-10-16T19:08:05.820+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-02T00:00:00+00:00 [queued]>
[2023-10-16T19:08:05.828+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-02T00:00:00+00:00 [queued]>
[2023-10-16T19:08:05.828+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 3
[2023-10-16T19:08:05.844+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): trusted> on 2023-10-02 00:00:00+00:00
[2023-10-16T19:08:05.850+0000] {standard_task_runner.py:57} INFO - Started process 335 to run task
[2023-10-16T19:08:05.854+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'wikipedia_etl', 'trusted', 'scheduled__2023-10-02T00:00:00+00:00', '--job-id', '6', '--raw', '--subdir', 'DAGS_FOLDER/wikipedia_etl.py', '--cfg-path', '/tmp/tmpuzs7qil_']
[2023-10-16T19:08:05.857+0000] {standard_task_runner.py:85} INFO - Job 6: Subtask trusted
[2023-10-16T19:08:05.899+0000] {task_command.py:415} INFO - Running <TaskInstance: wikipedia_etl.trusted scheduled__2023-10-02T00:00:00+00:00 [running]> on host 3522330161cc
[2023-10-16T19:08:05.966+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='wikipedia_etl' AIRFLOW_CTX_TASK_ID='trusted' AIRFLOW_CTX_EXECUTION_DATE='2023-10-02T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-10-02T00:00:00+00:00'
[2023-10-16T19:08:53.512+0000] {python.py:194} INFO - Done. Returned value was: None
[2023-10-16T19:08:53.528+0000] {taskinstance.py:1398} INFO - Marking task as SUCCESS. dag_id=wikipedia_etl, task_id=trusted, execution_date=20231002T000000, start_date=20231016T190805, end_date=20231016T190853
[2023-10-16T19:08:53.559+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2023-10-16T19:08:53.584+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
